### PR TITLE
Download and unpack Project

### PR DESCRIPTION
**Pull request by AlbertDeFusco**
This adds `anaconda-project download <username>/<project>`. You need not be logged in to download.

```
>anaconda-project download --help
usage: anaconda-project download [-h] [--directory PROJECT_DIR] [--no-unpack]
                                 [--parent_dir PARENT_DIR] [-s SITE]
                                 [-t TOKEN] [-u USERNAME]
                                 project

positional arguments:
  project               The project to download as <username>/<projectname>.
                        If <projectname>has spaces inclose everything in
                        quotes "<username>/<project name>".

optional arguments:
  -h, --help            show this help message and exit
  --directory PROJECT_DIR
                        Project directory containing anaconda-project.yml
                        (defaults to current directory)
  --no-unpack           Do not unpack the project archive.
  --parent_dir PARENT_DIR
                        Download archive to specific directory, otherwise
                        downloaded to current working directory.
  -s SITE, --site SITE  Select site to use
  -t TOKEN, --token TOKEN
                        Auth token or a path to a file containing a token
  -u USERNAME, --user USERNAME
                        User account, defaults to the current user
```

### CODE REVIEW

File: anaconda_project/client.py [hunk 1]
```diff
@@ -145,6 +160,30 @@ def upload(self, project_info, archive_filename, uploaded_basename, private):
 
         return res.json()
 
+    def download(self, project, project_dir=None, parent_dir=None):
+        """Download project archive and extract."""
+        owner, project_name = project.split('/')
```
Comment (mcg1969) [inline anaconda_project/client.py]: It seems like we should support the case when the owner is not supplied, by defaulting to the logged-in user. Thoughts? After all, you do already have `-u USERNAME` as an option

Comment (AlbertDeFusco) [reply anaconda_project/client.py]: done

**Comment by mcg1969** [comment]
Rebased off of master.

**Comment by mcg1969** [comment]
Finally got tests unstuck (which had nothing to do with this or other PRs). Thanks for this!

**Pull request merged**